02-KernelModules_CharacterDevice/Makefile

```
@@ -0,0 +1,18 @@
+#
+# character device
+#
+
+ifneq ($(KERNELRELEASE),)
+
+obj-m += concmod.o
+
+else
+
+.PHONY: all clean
+all:
+	${MAKE} -C ${BUILD_KERNEL} M=${CURDIR} modules
+
+clean:
+	${MAKE} -C ${BUILD_KERNEL} M=${CURDIR} clean
+
+endif
```

```suggestion
endif

```

Please add empty line at the end of file

---

add empty line at the end of file